config/validateRelease.py

```diff
@@ -80,9 +81,18 @@
 with tarfile.open( args.archive, "r:gz" ) as a:
 
 	# getmember still reads the whole archive, so might as well grab them all
-	# as we go. We need to strip the first directory from all paths too as that
-	# contains the release name
-	archivePaths = { os.path.join( *m.name.split( os.sep )[1:] ) for m in a.getmembers() if os.sep in m.name }
+	# as we go. We need to strip the first directory from all paths as that
+	# contains the release name.
+
+	archivePaths = sets.Set()
+
+	for m in a.getmembers() :
+		# ignore anything not under the release directory
+		if os.sep not in m.name :
+			continue
+		# Strip the empty component at the end for directories
+		parts = [ p for p in m.name.split( os.sep )[1:] if p ]
+		archivePaths.add( os.path.join( *parts ) )
```

Comment:
I think this is what `os.path.normpath()` is for.

Reply:
Ahh I forgot about that one - thats perfect, ta! https://github.com/GafferHQ/gaffer/commit/b04ce8615ca9b51406534da83b0b3fc32ac3acd7